### ISSUE TITLE
API Catalog

### ISSUE THREAD

**Issue by varshaprasad96** [kcp-dev/catalog#7]
This issue tracks the APICatalog epic which aims at providing a catalog of APIExports for users to be able to bind. More details on the design and approach can be found in the documentation: https://hackmd.io/5KcilJjjSfm24v3QWyjfBQ?view

Initial Prototype work:
- [ ] Finalize the design for `CatalogEntry` API.
- [ ] Design catalog controller which watches CEs, and validates whether the referenced APIExport exist.
- [ ] Design cli commands that allow users to bind to catalogs, list APIExports present in the catalog (based on hierarchy of workspaces), create CEs.

Future TODOs:
- [ ] Design and finalize lifecycle of CatalogEntry. 
- [ ] Fix the permission structure required to allow users bind to catalog.

**Comment by varshaprasad96** [comment]
cc: @dinhxuanvu

**Comment by pweil-** [comment]
@varshaprasad96 @dinhxuanvu Does this need a review on the breakdown of remaining tasks?  What is left to close out the MVP for an API Catalog?  Thanks!

**Comment by varshaprasad96** [comment]
@pweil- Have updated the tasks with the open PRs which are being worked on.

**Comment by pweil-** [comment]
Thanks @varshaprasad96. 🥇  I'm assuming that `Design and finalize lifecycle of CatalogEntry` represents a fairly significant portion of work and needs discussion.  A few questions:

1. Does this tie in at all with @ncdc 's API evolution R&D?
2. Should we begin coordinating this topic at the community calls now if it's part of MVP?

@gallettilance - FYI, relevant to your interests
cc @stevekuznetsov and @nrb who have been thinking about developer experience topics as well 👍

**Comment by varshaprasad96** [comment]
@pweil- Yes, the lifecycle of Catalog Entry is an extended task which needs more discussion. For now, we are trying to implement and close out the other 3 tasks to get the basic feature out of the door and get more user reviews on it. The design doc for the first phase as discussed is here: https://hackmd.io/5KcilJjjSfm24v3QWyjfBQ?view
